Game/utils/stringutil.cpp

```diff
@@ -0,0 +1,72 @@
+#include "stringutil.h"
+
+#include <locale>
+
+
+std::string StringUtil::toLower(const std::string& str) {
+  std::string ret(str);
+  
+  for(std::string::size_type i = 0; i < ret.length(); i++)
+  {
+    ret[i] = std::tolower(ret[i], std::locale::classic());
```

Comment:
Behavior of  `tolower` is undefined for non ancii-z input. Meanwhile, `str` is utf8 string.

Reply:
std::string stores the data as char, so on most platforms it holds 8bit, but as I understand it will still work with utf8, because of the way the data is encoded.

But you are probably right that the std::tolower will not do what we want it to do.
Did the first few lines of the recognice function do a to upper or lower by adding and subtracting 'A' and 'a'?
And this not also be utf8 incompatible?